pkg/reconcilers/base_reconciler.go

```diff
@@ -31,6 +31,44 @@ func CreateOnlyMutator(existing, desired common.KubernetesObject) (bool, error)
 	return false, nil
 }
 
+// Remove 'ThreeScaleApicastRequestTime' alert
+func RemovePrometheusRulesMutator(existing, desired common.KubernetesObject) (bool, error) {
+	existingPrometheusRule := existing.(*monitoringv1.PrometheusRule)
+	removed := false
+	updatedRules := []monitoringv1.Rule{}
+	group := existingPrometheusRule.Spec.Groups[0]
+
+	for _, rule := range group.Rules {
+		if rule.Alert != "ThreescaleApicastRequestTime" {
+			updatedRules = append(updatedRules, rule)
+		} else {
+			removed = true
+		}
+	}
+	if removed {
+		group.Rules = updatedRules
+		existingPrometheusRule.Spec.Groups[0] = group
+
+		log.Info("Alert 'ThreescaleApicastRequestTime' removed from PrometheusRules")
+		return true, nil
+	}
+	log.Info("Alert 'ThreescaleApicastRequestTime' not found, no update required.")
+	return false, nil
+}
+
+func UpdatePrometheusRulesMutator(existing, desired common.KubernetesObject) (bool, error) {
+	created, err := CreateOnlyMutator(existing, desired)
```

Comment:
`CreateOnlyMutator` is special mutator that will never update the existing resource, that is why it is called "CreateOnlyMutator". You can safely remove it from this function.